howtos/diffs/checkpointing.diff

```diff
@@ -10,17 +10,17 @@ index 63af6e8..59c0d1c 100644
  
  class CNN(nn.Module):
    """A simple CNN model."""
-@@ -166,6 +167,9 @@ def train_and_evaluate(model_dir: str, num_epochs: int, batch_size: int,
+@@ -165,6 +166,9 @@ def train_and_evaluate(config: ml_collections.ConfigDict, model_dir: str):
    model = create_model(init_rng)
-   optimizer = create_optimizer(model, learning_rate, momentum)
+   optimizer = create_optimizer(model, config.learning_rate, config.momentum)
  
 +  # Restore saved model
```

Comment:
How hard was it to figure out how to fix the HOWTO?

Reply:
It took me 5 mins to figure out how the HOWTOs were generated. Read up `git apply`, then it was easy from there. 

But I had one thing to say about git apply, the error message doesn't help at all. It could have been a bit more verbose and provide few more details.

Reply:
I also updated diffs recently and tried to improve the documentation a little bit - feel free to comment additional ideas to #473